logsrvd/sendlog.c

```diff
@@ -1167,7 +1300,36 @@ main(int argc, char *argv[])
     sock = connect_server(host, port);
     if (sock == -1)
 	goto bad;
-    printf("connected to %s:%s\n", host, port);
+#if defined(HAVE_OPENSSL)
+    if (tls) {
+        int ret;
+        if ((ssl_ctx = init_tls_client_context(ca_bundle, cert, key)) == NULL) {
+            printf("Unable to initialize ssl context\n");
+            goto bad;
+        }
+        if ((ssl = SSL_new(ssl_ctx)) == NULL) {
+            printf("Unable to allocate ssl object\n");
+            goto bad;
+        }
+        if (SSL_set_fd(ssl, sock) <= 0) {
+            printf("Unable to attach socket to the ssl object\n");
+            goto bad;
+        }
+        ret = SSL_ERROR_NONE;
+
+        if ((ret = SSL_connect(ssl)) != 1) {
+            printf("SSL_connect failed; ret=%d ssl_error=%d, stack=%s\n",
+                ret,
+                SSL_get_error(ssl, ret),
+                ERR_error_string(ERR_get_error(), NULL));
+            goto bad;
+        }
+
```

Comment:
Please use sudo_warnx() for these warnings instead of printf().

Reply:
fixed